src/utils/index.ts

```diff
@@ -24,3 +24,11 @@ export const storage = {
   set,
   get,
 };
+
+export const dotProduct = (vectorA: number[], vectorB: number[]): number => {
+  const vectorDistance = vectorA.reduce(
```

Comment:
We could check the lengths match of the vectors here.

Reply:
This has to return a number so I did the check in `engine/condition.ts` instead